tests/jballerina-integration-test/src/test/resources/websocket/src/wsservices/32_readyon_connect_retry.bal

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/io;
+
+@http:WebSocketServiceConfig {
+}
+service on new http:Listener(21034) {
+
+    resource function onOpen(http:WebSocketCaller wsEp) {
+        http:WebSocketClient wsClientEp = new("ws://localhost:15300/websocket", { callbackService:
+            readyOnConnectService, readyOnConnect: true, retryConfig: {}});
+        wsEp.setAttribute(ASSOCIATED_CONNECTION, wsClientEp);
+        wsClientEp.setAttribute(ASSOCIATED_CONNECTION, wsEp);
```

Comment:
First understand why ready is set to false in proxy scenarios. Understand the purpose of readyOnConnect config: https://riyafa.wordpress.com/2019/01/06/demystifying-websocket-in-ballerina-websocketlistener-websocketcaller-websocketclient/

Setting the ready to true is an incorrect way of writing a proxy

Reply:
Here I tested the negative test case for ready therefore I set the ready to true.